reviews/api/views.py

```diff
@@ -405,6 +405,38 @@ def get_queryset(self):
         return [value for key, value in reviews.items()]
 
 
+class InRevisionApiView(BaseReviewApiView):
+    """Return reviews for proposals which have been sent back for revision,
+    but for which a revision has not yet been submitted"""
+
+    default_sort = ('date_start', 'desc')
+    group_required = [settings.GROUP_SECRETARY]
+
+    def get_queryset(self):
+        # 1. Find reviews of revisions:
+        # A proposal not having one of these means
+        # that the review of its parent is "in revision"
+        revision_reviews = Review.objects.filter(
+            proposal__is_revision=True,    # Not a copy
+            stage__gte=Review.ASSIGNMENT,  # Not a supervisor review
+        )
+        # 2. Get candidate reviews:
+        # All reviews whose conclusion is "revision necessary"
+        candidates = Review.objects.filter(
+            stage=Review.CLOSED,
+            continuation=Review.REVISION,
+        )
```

Comment:
For readability I'd suggest swapping these qs arround; first get the dataset you're going to filter (and return), then the dataset that will be used to filter.

I instinctively go for that order, and it threw me off for a second that it's reversed.

Reply:
Yeah I started out that way, and ended up greatly preferring this. It's probably personal preference.